tests/hazmat/primitives/test_ssh.py

```diff
@@ -1236,5 +1241,330 @@ def test_public_bytes(self, backend):
             mode="rb",
         )
         cert = load_ssh_public_identity(data)
-        assert isinstance(cert, ssh.SSHCertificate)
+        assert isinstance(cert, SSHCertificate)
         assert data == cert.public_bytes()
+
+
+class TestSSHCertificateBuilder:
+    def test_builds_a_cert(self):
+        private_key = ec.generate_private_key(ec.SECP256R1())
+        public_key = ec.generate_private_key(ec.SECP256R1()).public_key()
+        valid_before = datetime.datetime(2023, 1, 16, 18, 43)
+        tz = pytz.timezone("US/Eastern")
+        valid_before = tz.localize(valid_before)
+        utc_time_before = datetime.datetime(2023, 1, 16, 23, 43)
+        valid_after = datetime.datetime(2023, 7, 16, 22, 43)
+        key_id = b"test"
+        valid_principals = [b"eve", b"alice"]
+        builder = (
+            SSHCertificateBuilder()
+            .public_key(public_key)
+            .type(SSHCertificateType.USER)
+            .valid_before(valid_before)
+            .valid_after(valid_after)
+            .key_id(key_id)
+            .valid_principals(valid_principals)
+            .add_critical_option(b"ordered", b"")
+            .add_critical_option(b"maybe", b"test2")
+            .add_extension(b"test", b"a value")
+            .add_extension(b"allowed", b"")
+        )
+        cert = builder.sign(private_key)
+        cert.verify_cert_signature()
+        cert_public_key = cert.public_key()
+        assert isinstance(cert_public_key, ec.EllipticCurvePublicKey)
+        assert cert_public_key.public_numbers() == public_key.public_numbers()
+        assert cert.serial == 0
+        assert cert.type is SSHCertificateType.USER
+        assert cert.key_id == key_id
+        assert cert.valid_principals == valid_principals
+        assert cert.valid_before == utc_time_before
+        assert cert.valid_after == valid_after
+        assert cert.critical_options == {b"ordered": b"", b"maybe": b"test2"}
+        assert list(cert.critical_options.keys()) == [b"maybe", b"ordered"]
+        assert cert.extensions == {b"test": b"a value", b"allowed": b""}
+        assert list(cert.extensions.keys()) == [b"allowed", b"test"]
```

Comment:
```suggestion
        assert list(cert.extensions) == [b"allowed", b"test"]
```

same for all usages of `keys`